src/Raven.Server/Commercial/LetsEncrypt/LetsEncryptSimulationHelper.cs

```diff
@@ -0,0 +1,178 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Connections;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.DependencyInjection;
+using Raven.Server.Config;
+using Raven.Server.Https;
+using Raven.Server.ServerWide;
+using Sparrow.Logging;
+using Sparrow.Platform;
+
+namespace Raven.Server.Commercial.LetsEncrypt;
+
+public class LetsEncryptSimulationHelper
+{
+    internal static readonly Logger Logger = LoggingSource.Instance.GetLogger<LicenseManager>("Server");
+
+    internal class UniqueResponseResponder : IStartup
+    {
+        private readonly string _response;
+
+        public UniqueResponseResponder(string response)
+        {
+            _response = response;
+        }
+
+        public IServiceProvider ConfigureServices(IServiceCollection services)
+        {
+            return services.BuildServiceProvider();
+        }
+
+        public void Configure(IApplicationBuilder app)
+        {
+            app.Run(async context =>
+            {
+                await context.Response.WriteAsync(_response);
+            });
+        }
+    }
+     public static async Task SimulateRunningServer(ServerStore serverStore, X509Certificate2 serverCertificate, string serverUrl, string nodeTag,
+        IPEndPoint[] addresses, int port, string settingsPath, SetupMode setupMode, CancellationToken token)
+    {
+        var configuration = RavenConfiguration.CreateForServer(null, settingsPath);
+        configuration.Initialize();
+        var guid = Guid.NewGuid().ToString();
+
+        IWebHost webHost = null;
+        try
+        {
+            try
+            {
+                var responder = new UniqueResponseResponder(guid);
+
+                var webHostBuilder = new WebHostBuilder()
+                    .CaptureStartupErrors(captureStartupErrors: true)
+                    .UseKestrel(options =>
+                    {
+                        var httpsConnectionMiddleware = new HttpsConnectionMiddleware(serverStore.Server, options);
+                        httpsConnectionMiddleware.SetCertificate(serverCertificate);
+
+                        if (addresses.Length == 0)
+                        {
+                            var defaultIp = new IPEndPoint(IPAddress.Parse("0.0.0.0"), port == 0 ? 443 : port);
```

Comment:
Don't we already have const for "0.0.0.0"? We could use it here

Reply:
Now it's within `Client.Constants`